cli/running/lua/launcher.lua

```diff
@@ -12,13 +12,86 @@
 -- Several useful links:
 -- https://porkmail.org/era/unix/arg-max.html
 -- https://unix.stackexchange.com/a/120842
-
+local fun   = require('fun')
 local os = require('os')
 local console = require('console')
 local log = require('log')
 local title = require('title')
+local fio = require('fio')
+
+local iter, op  = fun.iter, fun.operator
+
+local function mkdir(dirname)
+    if not fio.mkdir(dirname, tonumber('0750', 8)) then
+        log.error("Can't mkdir %s: %s", dirname)
+    os.exit(1)
+    end
+end
+
+--- Makes default dirs for wal, memtx and vinyl files.
+local function mk_default_dirs()
+    local init_dirs = {
+            os.getenv('TT_WAL_DIR'),
+            os.getenv('TT_VINYL_DIR'),
+            os.getenv('TT_MEMTX_DIR'),
+        }
+    for _, dir in ipairs(init_dirs) do
+        if fio.stat(dir) == nil then
+            mkdir(dir)
+        end
+    end
+end
+
+local function reducer(res, l, r)
+    if res ~= nil then
+        return res
+    end
+    if tonumber(l) == tonumber(r) then
+        return nil
+    end
+    return tonumber(l) > tonumber(r)
+end
 
 
+local function split(self, sep)
+    local sep, fields = sep or ":", {}
+    local pattern = string.format("([^%s]+)", sep)
+    self:gsub(pattern, function(c) table.insert(fields, c) end)
+    return fields
+end
+
+local function split_version(version_string)
+    local vtable  = split(version_string, '.')
+    local vtable2 = split(vtable[3],  '-')
+    vtable[3], vtable[4] = vtable2[1], vtable2[2]
+    return vtable
+end
+--- Returns if version of tarantool is greater then expected.
```

Comment:
Somewhere you use two blank lines between the function definition, somewhere one, somewhere none. Please make your choice.